packages/nimble-components/src/table-column/base/cell-view/index.ts

```diff
@@ -29,6 +29,14 @@ export abstract class TableCellView<
     @observable
     public recordId?: string;
 
+    /**
+     * Gets the child elements in this cell view that should be able to be reached via Tab/ Shift-Tab,
+     * if any.
+     */
+    public get tabbableChildren(): HTMLElement[] {
```

Comment:
It's called whenever a row (or a cell in a particular row) is focused and a key is pressed, and KeyboardNavigationManager is determining how to move focus.

I'd say that the current expectation is that the set of elements provided here is fixed. KeyboardNavigationManager doesn't cache elements internally so the main expectation is that `tabbableChildren.length` isn't changing (since KeyboardNavigationManager represents focused cellContent as `rowIndex=X, columnIndex=Y, cellContentIndex=Z`, `cellContentIndex` could be out-of-bounds if `tabbableChildren` shrank to a smaller array size.
If we had any expectation of needing to support this being more dynamic the logic in KeyboardNavigationManager might need tweaks to reset focus back to the cell if it changes or something like that.

Reply:
Making it observable seems like a good idea. I'll leave this thread open until I create a followup work item.

Reply:
#2210